scheduler/src/cook/postgres.clj

```diff
@@ -0,0 +1,98 @@
+(ns cook.postgres
+  (:require [clojure.tools.logging :as log]
+            [clojure.java.jdbc :as sql]
+            [cook.config :as config])
+  (:import (java.lang Runtime)
+           (java.util Random)))
+
+(defn setup-database
+  "Setup a fresh cook schema from scratch. At present, calls out
+  to a bash script, passing it the target schema name and hopes it does the right thing."
+  [setup-script schema-name]
+  ; This environmental variable contains the path of a setup script that, given a schema name, creates a database.
+  (let [^Process p (-> (Runtime/getRuntime)
+                       (.exec (str setup-script " " schema-name)))
+        _ (log/info "Waiting for database schema production process" (.pid p) "to exit")
+        exit-code (.waitFor p)]
+    (assert (= exit-code 0) (str "Exit code for process creation is" exit-code "not zero"))))
+
+(defn make-database-connection-dictionary-for-unit-tests
```

Comment:
Shouldn't this function be under `scheduler/test/` instead of `scheduler/src/`?

Reply:
I'd like to localize all postgres functions into this one file. Also, this code is going to need to refactor (in semi-uncertain ways at the moment) when we add in connection pooling. I'd like to keep it here for now.

Reply:
Why would we keep unit-test-specific stuff in `src`?

Reply:
Thinking on it further, this is used in the database configuration code, so this is called indirectly by pg-db, so therefore needs to be here (in in a file this namespace depends on) in order to be called.